pkg/restore/change_storageclass_action_test.go

```diff
@@ -0,0 +1,394 @@
+				Spec: corev1api.PersistentVolumeSpec{
+					StorageClassName: "storageclass-1",
+				},
+			},
+			configMap: &corev1api.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "velero",
+					Name:      "change-storage-class",
+					Labels: map[string]string{
+						"velero.io/plugin-config":        "true",
+						"velero.io/change-storage-class": "RestoreItemAction",
+					},
+				},
+				Data: map[string]string{
+					"storageclass-3": "storageclass-4",
+				},
+			},
+			storageClass: &storagev1api.StorageClass{
+				ObjectMeta: objectMeta("", "storageclass-2"),
+			},
+			want: &corev1api.PersistentVolume{
+				ObjectMeta: objectMeta("velero", "pv-1"),
+				Spec: corev1api.PersistentVolumeSpec{
+					StorageClassName: "storageclass-1",
+				},
+			},
+		},
+		{
+			name: "when persistent volume claim's storage class has no mapping in the config map, the item is returned as-is",
+			pvOrPVC: &corev1api.PersistentVolumeClaim{
+				ObjectMeta: objectMeta("velero", "pv-1"),
+				Spec: corev1api.PersistentVolumeClaimSpec{
+					StorageClassName: stringPtr("storageclass-1"),
+				},
+			},
+			configMap: &corev1api.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "velero",
+					Name:      "change-storage-class",
+					Labels: map[string]string{
+						"velero.io/plugin-config":        "true",
+						"velero.io/change-storage-class": "RestoreItemAction",
+					},
+				},
+				Data: map[string]string{
+					"storageclass-3": "storageclass-4",
+				},
+			},
+			storageClass: &storagev1api.StorageClass{
+				ObjectMeta: objectMeta("", "storageclass-2"),
+			},
+			want: &corev1api.PersistentVolumeClaim{
+				ObjectMeta: objectMeta("velero", "pv-1"),
+				Spec: corev1api.PersistentVolumeClaimSpec{
+					StorageClassName: stringPtr("storageclass-1"),
+				},
+			},
+		},
+		{
+			name: "when persistent volume's storage class is mapped to a nonexistent storage class, an error is returned",
+			pvOrPVC: &corev1api.PersistentVolume{
+				ObjectMeta: objectMeta("velero", "pv-1"),
+				Spec: corev1api.PersistentVolumeSpec{
+					StorageClassName: "storageclass-1",
+				},
+			},
+			configMap: &corev1api.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "velero",
+					Name:      "change-storage-class",
+					Labels: map[string]string{
+						"velero.io/plugin-config":        "true",
+						"velero.io/change-storage-class": "RestoreItemAction",
+					},
+				},
+				Data: map[string]string{
+					"storageclass-1": "nonexistent-storage-class",
+				},
+			},
+			storageClass: &storagev1api.StorageClass{
+				ObjectMeta: objectMeta("", "storageclass-2"),
+			},
+			wantErr: errors.New("error getting storage class nonexistent-storage-class from API: storageclasses.storage.k8s.io \"nonexistent-storage-class\" not found"),
+		},
+		{
+			name: "when persistent volume claim's storage class is mapped to a nonexistent storage class, an error is returned",
+			pvOrPVC: &corev1api.PersistentVolumeClaim{
```

Comment:
Is there a reason you're not using the `test` package's constructors for these?

Reply:
No, good call - updated.